ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -286,6 +286,7 @@ def _create_session_core(
         self,
         session_constructor: Callable[[SessionInformation], TSession],
         instrument_type_id: str,
+        initialization_behavior: SessionInitializationBehavior = SessionInitializationBehavior.AUTO,
```

Comment:
Post-RFC, make this parameter required in the `_core` methods to ensure that we don't accidentally leave it out of any of the callers.